include/veriblock/pop/blockchain/base_block_tree.hpp

```diff
@@ -148,13 +140,7 @@ struct BaseBlockTree {
   }
 
   virtual bool loadTip(const hash_t& hash, ValidationState& state) {
-    auto* tip = getBlockIndex(hash);
-    if (!tip) {
-      return state.Invalid(block_t::name() + "-no-tip");
-    }
-
-    this->overrideTip(*tip);
-    return true;
+    return setState(hash, state);
```

Comment:
Why do you setState here?

Purpose of loadTip was to set tip for existing state, not applying loaded blocks

Reply:
I was looking for more general method of updating current tip instead of overrideTip. This PR is draft and not ready for a review yet.